lib/rust/ast/generation/src/api.rs

```diff
@@ -0,0 +1,672 @@
+//! This module exports jni bindings generator.
```

Comment:
You're missing `new` constructors all throughout this module, and a lot of your doc comments aren't proper sentences.

Additionally, doc comments should be wrriten in an _imperative_ form: `Generate the blahblah`, not `Generates the blahblah`.